docgen/src/gettingstarted.md

```diff
@@ -6,49 +6,323 @@ nav_groups:
+to type this command in your shell:
 
 ```sh
-npm install react-instantsearch --save
+npm install --save react-instantsearch
 ```
 
-app.js
-```js
+## Add the InstantSearch wrapper
+
+The instantsearch wrapper is the component that will connect to Algolia
+and that will synchronise all the widgets together. It maintains the state
+of the search, do the queries and provide the results to the widgets so
+that they can update themselves if needed.
+
+```javascript
 import React from 'react';
-import ReactDOM from 'react-dom';
 
-import {
-  InstantSearch,
-  SearchBox,
-  Hits,
-} from 'react-instantsearch';
+import {InstantSearch} from 'react-instantsearch';
 
-const App = () =>
+var App = () =>
   <InstantSearch
-    appId="latency"
-    apiKey="6be0576ff61c053d5f9a3225e2a90f76"
-    indexName="movies"
+    appId={appId}
+    apiKey={apiKey}
+    indexName={indexName}
   >
-    <div>
-      <SearchBox />
+   // The instantsearch components will go here
+  </InstantSearch>
+
+// Needed only if you're js app doesn't do it already.
+// Create-react-app does it for you ;)
+React.render(<App />, document.querySelector('#app'));
+```
+
+As you can see, there are a few mandatory props. Those props are the
+credentials of your application in Algolia. They can be found in your Algolia
+dasboard.
+
+Congratulations, your application is now connected to Algolia. This wrapper
+will provide a context so that the instantsearch components you add inside
+can interact with the search.
+
+In this section we've seen:
+ - how to connect a part of a react application to Algolia
+ - configure your credentials with react-instantsearch
+
+ > To get more *under the hood* information about the `InstantSearch` wrapper
+ > component, we have [full documentation](InstantSearch.html) in the widget
+ > section.
+
+## Display the results
+
+The core of a search is to display results. By default, react-instantsearch
+will do a query at the start of the page, because the relevance from the
+Algolia engine will always sort your records even though they are not
+filtered yet.
+
+Therefore the next widget we want to use is the Hits widget. This widget will
+display all the results returned by Algolia, and it will update when there
+are new results.
+
+And while we're doing that, let's create a new component for the search:
+
+```javascript
+// First, we need to add the Hits component to our import
+import {InstantSearch, Hits, SearchBox} from 'react-instantsearch';
+
+// ... We skip the rest of the code here
+
+function Search() {
+  return (
+    <div className='container'>
       <Hits />
     </div>
-  </InstantSearch>;
-};
+  );
+}
+```
+
+We can then call this component in the render of App:
+
+```javascript
+var App = () =>
+  <InstantSearch
+    appId={appId}
+    apiKey={apiKey}
+    indexName={indexName}
+  >
+   <Search/>
+  </InstantSearch>
+```
+
+You should now be able to see the results... in a very raw way. This
```

Comment:
Ellipsis _can_ be seen as an hesitation from the reader, it's easier to not use them most of the time.